rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/WorkflowSubmitter.java

```diff
@@ -69,7 +73,15 @@ public WorkflowSubmitter(Scheduler scheduler) {
     public JobId submit(File workflowFile, Map<String, String> variables) throws NotConnectedRestException,
             PermissionRestException, SubmissionClosedRestException, JobCreationRestException {
         try {
-            return scheduler.submit(createJobObject(workflowFile, variables));
+            Job job = createJobObject(workflowFile, variables);
+            JobId jobId = scheduler.submit(job);
+            // Create Job's SVG visualization file
+            File visualizationFile = new File(PortalConfiguration.jobIdToPath(jobId.value()) + ".html");
+            Files.deleteIfExists(visualizationFile.toPath());
+            FileUtils.write(new File(visualizationFile.getAbsolutePath()),
+                            job.getVisualization(),
```

Comment:
the visualization might not be present, you need to check this before writing the file